src/ts/common.ts

```diff
@@ -95,6 +95,24 @@ const FavTagC = new FavTag();
 const TodolistC = new Todolist();
 
 const TableSortingC = new TableSorting();
+// for searching inputs, allow specific triggers for East & South East Asian characters
+const hasEastSEAsian = (s: string): boolean => (
+  /\p{Script=Han}|\p{Script=Hiragana}|\p{Script=Katakana}|\p{Script=Hangul}|\p{Script=Thai}|\p{Script=Lao}|\p{Script=Khmer}|\p{Script=Myanmar}/u.test(s)
+);
+
+// helper to count how many characters in the string are visible characters (grapheme)
+let graphemeSegmenter: Intl.Segmenter | null = null;
+if (typeof Intl !== 'undefined' && Intl.Segmenter) {
+  graphemeSegmenter = new Intl.Segmenter(undefined, { granularity: 'grapheme' });
+}
+
+const countGraphemes = (text: string): number => {
+  if (!graphemeSegmenter) {
```

Comment:
why not create the graphemeSegmenter in this function instead of having it as a global var?